### PR TITLE
Assuming roles

### PR DESCRIPTION
**Pull request by PaulCampbell**
@joshski

### CODE REVIEW

File: TTC.Deployment.Tests/RoleHelper.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
+using System;
+using Amazon.IdentityManagement.Model;
+using Amazon.IdentityManagement;
+using Amazon.Runtime;
+
+namespace TTC.Deployment.Tests
+{
+    internal class RoleHelper
```
Comment (PaulCampbell) [inline TTC.Deployment.Tests/RoleHelper.cs]: So - I'm not mad keen on this helper.

Perhaps a user factory would be better?

GiveMeAUser.WhoCanAssumeS3Role
GiveMeAUser.WhoCanAssumeGodRole

Or something else - not sure.  Only thing I know is its intent currently isn't that clear

**Comment by PaulCampbell** [comment]
Oh, Dont merge yet... Waiting on test run

File: TTC.Deployment.Tests/RoleHelper.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                }".Replace("{RoleARN}", role.Arn)
+            });
+
+            var accessKey = _iamClient.CreateAccessKey(new CreateAccessKeyRequest { UserName = user.UserName }).AccessKey;
+            var userCredentials = new BasicAWSCredentials(accessKey.AccessKeyId, accessKey.SecretAccessKey);
+            return role;
+
+        }
+
+        public void DeleteRole(string roleName)
```
Comment (PaulCampbell) [inline TTC.Deployment.Tests/RoleHelper.cs]: All this clean up crap should probably go in a `CleanerUpper` or something

**Comment by PaulCampbell** [comment]
Frankly this whole "test throws, stuff not cleaned up" thing is driving me a bit nuts.

Needs a solution i think?

**Comment by joshski** [comment]
Yeah I can't see why the push should need to create a role. Let's try without it when we get to green again :)

**Comment by daniel-baptista-travcorp** [comment]
So we can merge this?

**Comment by PaulCampbell** [comment]
No :( I still have a few failing tests. trying to sort them now

File: TTC.Deployment.AmazonWebServices/Deployer.cs [hunk 3]
```diff
@@ -164,7 +168,7 @@ private void EnsureCodeDeployRoleExists()
 
             _iamClient.PutRolePolicy(new PutRolePolicyRequest
             {
-                RoleName = _awsConfiguration.RoleName,
+                RoleName = _awsConfiguration.RoleArn,
```
Comment (Jetski5822) [inline TTC.Deployment.AmazonWebServices/Deployer.cs]: Shouldnt this actually be RoleName?

**Comment by Jetski5822** [comment]
no longer required

**Pull request closed without merging**